tests/test_cuda_bd_exchange_mover.py

```diff
@@ -16,10 +18,77 @@
 from timemachine.md import builders
 from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.md.exchange.exchange_mover import BDExchangeMove as RefBDExchangeMove
-from timemachine.potentials import HarmonicBond, Nonbonded
+from timemachine.md.minimizer import check_force_norm
+from timemachine.potentials import HarmonicBond, Nonbonded, SummedPotential
 from timemachine.testsystems.relative import get_hif2a_ligand_pair_single_topology
 
 
+def verify_bias_deletion_moves(
+    mol_groups: List,
+    bdem,
+    ref_bdem: RefBDExchangeMove,
+    conf: NDArray,
+    box: NDArray,
+    total_num_proposals: int,
+    proposals_per_move: int,
+    rtol: float,
+    atol: float,
+):
+    assert bdem.last_log_probability() == 0.0, "First log probability expected to be zero"
+    accepted = 0
+    last_conf = conf
+    for step in range(total_num_proposals // proposals_per_move):
+        x_move, x_box = bdem.move(last_conf, box)
+        # The box will never change
+        np.testing.assert_array_equal(box, x_box)
+        num_moved = 0
+        new_pos = None
+        idx = -1
+        for i, mol_idxs in enumerate(ref_bdem.water_idxs_np):
+            if not np.all(x_move[mol_idxs] == last_conf[mol_idxs]):
+                num_moved += 1
+                new_pos = x_move[mol_idxs]
+                idx = i
+        if num_moved > 0:
+            accepted += 1
+        if num_moved == 1:
+            # The molecules should all be imaged in the home box
+            np.testing.assert_allclose(image_frame(mol_groups, x_move, x_box), x_move)
+            # Verify that the probabilities and per mol energies agree when we do accept moves
+            # can only be done when we only attempt a single move per step
+            before_log_weights = ref_bdem.batch_log_weights(last_conf, box)
+            after_log_weights, tested = ref_bdem.batch_log_weights_incremental(
+                last_conf, x_box, idx, new_pos, before_log_weights
+            )
+            np.testing.assert_array_equal(tested, x_move)
```

Comment:
(Deleted the old comment) - I'm not sure actually why `batch_log_weights_incremental` returns the a `new_conf` in addition to the weights, since it will always be simply `conf.at[water_idxs[idx]].set(new_pos)` - we can simply test that the *final* position in an arbitrary sequence of moves is the same.